src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('all sides of triangle must be greater than 0');
+    }
+
+    const biggestSide = Math.max(a, b, c);
+    const sumOfSmaller = a + b + c - biggestSide;
+
+    if (biggestSide >= sumOfSmaller) {
+      throw new Error(`sides ${a}, ${b} and ${c} can not form a triangle`);
+    }
+  }
+
+  public getArea(): number {
+    const semiperimeter = 0.5 * (this.a + this.b + this.c);
 
-export class Triangle implements Figure {}
+    return (
+      Math.floor(
+        Math.sqrt(
+          semiperimeter *
+            (semiperimeter - this.a) *
+            (semiperimeter - this.b) *
+            (semiperimeter - this.c),
+        ) * 100,
+      ) / 100
+    );
+  }
+}
+
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('radius of circle must be greater than 0');
+    }
+  }
+
+  public getArea(): number {
+    return Math.floor(Math.PI * this.radius * this.radius * 100) / 100;
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('all sides of rectangle must be greater than 0');
+    }
```

Comment:
you can create function like

```
function checkOnError(errorMessage ,...args: number[]): void {
  if (args.some((arg: number) => arg <= 0)) {
    throw new Error(errorMessage);
  }
}
```
It is needed because you have the same logic in three places.